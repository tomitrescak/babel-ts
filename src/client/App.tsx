import * as React from 'react';
import { ApolloProvider } from 'react-apollo';
import styled, { keyframes } from 'styled-components';

import logo from './logo.svg';

import { hot } from 'react-hot-loader';
import { BooksContainer } from './apollo_view';
import { client } from './config/apollo';

const Title = styled.h1`
  font-size: 1.5em;
  text-align: center;
  color: palevioletred;
`;

const Holder = styled.div`
  text-align: center;
`;

const rotate = keyframes`
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}`;

const Link = styled.a`
  color: #61dafb;
`;

const Header = styled.div`
  background-color: #282c34;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
`;

const Logo = styled.img`
  animation: ${rotate} infinite 3s ease-in-out;
  height: 40vmin;
`;

class App extends React.Component {
  public state = {
    i: 0
  };

  public render() {
    return (
      <ApolloProvider client={client}>
        <Holder>
          <If condition={true}>JSX Works</If>
          <Header>
            <Title>Other {this.state.i}</Title>
            <Logo src={logo} alt="logo" />
            <BooksContainer />
            <p>
              Edit <code>src/App.js</code> and save to reload.
            </p>
            <Link href="https://reactjs.org" target="_blank" rel="noopener noreferrer">
              Learn React
            </Link>
            <button onClick={this.increment}>Increment</button>
          </Header>
        </Holder>
      </ApolloProvider>
    );
  }

  private increment = () => {
    this.setState({ i: this.state.i + 1 });
  };
}

export default hot(module)(App);
