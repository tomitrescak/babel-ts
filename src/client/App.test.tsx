import * as React from 'react';
import * as ReactDOM from 'react-dom';
import * as renderer from 'react-test-renderer';

import App from './App';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('renders snapshot', () => {
  const component = renderer.create(<App />);
  const root = component.root;

  expect(component).toMatchSnapshot();

  // const button = root.findByType('button');
  // button.props.onClick();

  // expect(component).toMatchSnapshot();
});
