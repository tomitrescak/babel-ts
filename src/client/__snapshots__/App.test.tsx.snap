// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders snapshot 1`] = `
<div
  className="App__Holder-cZbTBl crTnHz"
>
  JSX Works
  <div
    className="App__Header-pclPN eCWRij"
  >
    <h1
      className="App__Title-ikgjvl gwMrQA"
    >
      Other 
      0
    </h1>
    <img
      alt="logo"
      className="App__Logo-cQOErG gcQhau"
      src="logo.svg"
    />
    <div>
      Loading ...
    </div>
    <p>
      Edit 
      <code>
        src/App.js
      </code>
       and save to reload.
    </p>
    <a
      className="App__Link-cevQCU doqOVW"
      href="https://reactjs.org"
      rel="noopener noreferrer"
      target="_blank"
    >
      Learn React
    </a>
    <button
      onClick={[Function]}
    >
      Increment
    </button>
  </div>
</div>
`;

exports[`renders snapshot 2`] = `
<div
  className="App__Holder-cZbTBl crTnHz"
>
  JSX Works
  <div
    className="App__Header-pclPN eCWRij"
  >
    <h1
      className="App__Title-ikgjvl gwMrQA"
    >
      Other 
      1
    </h1>
    <img
      alt="logo"
      className="App__Logo-cQOErG gcQhau"
      src="logo.svg"
    />
    <div>
      Loading ...
    </div>
    <p>
      Edit 
      <code>
        src/App.js
      </code>
       and save to reload.
    </p>
    <a
      className="App__Link-cevQCU doqOVW"
      href="https://reactjs.org"
      rel="noopener noreferrer"
      target="_blank"
    >
      Learn React
    </a>
    <button
      onClick={[Function]}
    >
      Increment
    </button>
  </div>
</div>
`;
